items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -57,142 +103,123 @@ class SierraItemUpdater(
-            ) =>
-          warn(s"Item lookup missing items: $missingItems")
-          itemsReturned
-            .map(item => {
-              val location = existingItems.get(item.id).flatten
-              item.id -> item.accessCondition(location)
-            })
-            .toMap
-        case Left(itemLookupError) =>
-          error(s"Item lookup failed: $itemLookupError")
-          Map.empty[SierraItemNumber, Option[DisplayAccessCondition]]
-      }
-
-      accessConditions = maybeAccessConditions
-        .collect { case (itemId, Some(ac)) => itemId -> ac }
-
-    } yield accessConditions
-
-  private def setAvailableDates(
-    item: DisplayItem
+  def setAvailableDates(
+    item: DisplayItem,
+    sierraItemLocation: SierraLocation
   ): Future[DisplayItem] = {
     // there is only ever one location per physicalItem and one accessCondition per location,
     // but this may sometimes be empty if it could not be fetched.
     val accessCondition = item.locations.headOption
       .flatMap(_.accessConditions.headOption)
 
+    val locationName = sierraItemLocation.code match {
+      case "harop" => "deepstore"
+      case _       => "library"
+    }
+
     if (accessCondition.exists(_.isRequestable)) {
-      getAvailableDates(accessCondition.head)
-        .map(availableDates => item.copy(availableDates = Some(availableDates)))
+      locationName match {
+        case "deepstore" =>
+          deepstoreItemAvailabilities(getVenuesOpeningTimes(locationName)).map(
+            availableDates =>
+              item.copy(
+                availableDates = Some(availableDates)
+            )
+          )
+        case "library" =>
+          libraryItemAvailabilities(getVenuesOpeningTimes(locationName)).map(
+            availableDates =>
+              item.copy(
+                availableDates = Some(availableDates)
+            )
+          )
+      }
     } else {
       Future.successful(item)
     }
   }
 
-  private def getAvailableDates(
-    accessCondition: DisplayAccessCondition
-  ): Future[List[AvailabilitySlot]] = {
-    val venueFromDisplayAccessCondition = Map(
-      "online-request" -> "library"
-      // other venue to be added as DisplayAccessMethod id -> content-api venue title
-    )
-
-    def getLeadTimeInDays(openingTimes: List[OpenClose]): Int = {
-      val timeAtVenue = LocalDateTime.now(venueClock)
-      val isWorkingDay = timeAtVenue.toLocalDate.isEqual(
-        LocalDate.parse(
-          openingTimes.head.open,
-          DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
-        )
-      )
-      accessCondition.method.id match {
-        case "online-request" if (timeAtVenue.getHour < 10 || !isWorkingDay) =>
-          1
-        case "online-request" if timeAtVenue.getHour >= 10 => 2
-      }
-    }
+  private def getVenuesOpeningTimes(
+    locationName: String
+  ): Future[Map[String, List[AvailabilitySlot]]] =
+    for {
+      venues <- venuesOpeningTimesLookup
```

Comment:
I'd avoid the `for` comprehension here as you have only one future to unwrap (and it'll unnest things a bit here).